extensions/amp-story/1.0/utils.js

```diff
@@ -231,6 +231,30 @@ export function getSourceOriginForElement(element, url) {
   return domainName;
 }
 
+/**
+ * Checks if document is being served from cache.
+ * @param {!Document} doc
+ * @return {boolean}
+ */
+export function isServedFromCache(doc) {
```

Comment:
Would that work without us having to write extra code?
https://github.com/ampproject/amphtml/blob/master/src/service/url-impl.js#L152

Alternatively, you could also get the absolute URL from the relative one even if not served from the cache. That should be a no-op.